resources/world/world_data.gd

```diff
@@ -329,6 +340,35 @@ func is_room_cell(p_index: int) -> bool:
 	return value
 
 
+func set_room(type: String, p_room_data: RoomData) -> void:
+	if not rooms.has(type):
+		rooms[type] = []
+	
+	rooms[type].append(p_room_data)
+
+
+# Returns an Array of all RoomData
+func get_all_rooms() -> Array:
+	var value := []
+	
+	for data_array in rooms.values():
+		value.append_array(data_array)
+	
+	return value
+
+
+# Returns an Array of RoomData from the specified "type" 
+func get_rooms_of_type(p_type: String) -> Array:
+	var value := []
+	
+	if rooms.has(p_type):
+		value = rooms[p_type]
+	else:
+		push_error("Type of room not found: %s. Available types: %s"%[p_type, rooms.keys()])
```

Comment:
Requesting an array of rooms of a type that isn't present in the map shouldn't be treated as an error (at least by this function). Not having rooms of a certain type is perfectly valid, and this should just return an empty array in that case. If the caller of this function expects the room type to exist, it can then treat it as an error if the array has a size of 0. Also, the Dictionary.get() function allows setting a default value to return if the key isn't present in the dictionary, which would simplify this function to `return rooms.get(p_type, Array())`